### PR TITLE
Adding Heroku deploy option

### PR DESCRIPTION
**Pull request by IvanGodinez21**
Users now can deploy to Heroku

Co-Authored-By: Josué B. Iglesias <54423138+JosueIglesias@users.noreply.github.com>

### CODE REVIEW

**Comment by IvanGodinez21** [comment]
@MarcDonald If you have any questions please @me, I will answer any questions you have about changes in certain files.

**Comment by IvanGodinez21** [comment]
I forgot to mention that you can test to deploy an app on Heroku using in this commit from my development repository
https://github.com/IvanGodinez21/twitch-spotify-request-bot/tree/3183dd6702454ceefe187f79bf30d9f9b4396bdf

Notes: If you want to test, you must use that commit from my repository, app.json specifies to use a different repository (In  the PR specifies to use master from this repository)

Basically, when the PR is accepted, the button in the README.md will work

**Comment by MarcDonald** [comment]
Hey, thanks for the PR.

I don't seem to be able to test the deploy using Heroku using that link you provided. Whenever I go to https://dashboard.heroku.com/new?template=https://github.com/IvanGodinez21/twitch-spotify-request-bot/tree/3183dd6702454ceefe187f79bf30d9f9b4396bdf I get the `No app.json located in the repo URL provided` error message.

The same error happens when I use the Deploy to Heroku button on your commit that links to https://dashboard.heroku.com/new?button-url=https%3A%2F%2Fgithub.com%2FIvanGodinez21%2Ftwitch-spotify-request-bot%2Ftree%2F3183dd6702454ceefe187f79bf30d9f9b4396bdf&template=https%3A%2F%2Fgithub.com%2FIvanGodinez21%2Ftwitch-spotify-request-bot%2Ftree%2Fheroku


image

**Comment by IvanGodinez21** [comment]
Are you sure you did it in the correct way?, @JosueIglesias and I were testing it before making the PR and it was working...

_This is a quick example, I created a new Heroku account_

https://user-images.githubusercontent.com/45635827/120118980-e352bd80-c15a-11eb-9a39-2aea0a1ca28f.mp4

**Comment by MarcDonald** [comment]
Did some more testing and it turns out one of my Chrome extensions ([this one](https://chrome.google.com/webstore/detail/clearurls/lckanjgmijmafbedllaakclkaicjfmnk)) was messing up the button's redirect link. Once I turned that extension off it all worked great.

Before I merge this in could you add an additional bullet point onto the Deploy section of the Readme with a point telling the user to set the Spotify app's redirect URL to `https://users-app-name.herokuapp.com/spotifyAuth`?

Thanks again for the PR it's pretty neat being able to deploy in basically one click 😄

**Comment by IvanGodinez21** [comment]
@MarcDonald
Don't worry. I added a new step to the README.md in Deploy section

**Comment by MarcDonald** [comment]
Excellent, thanks very much. I'll merge it in now

**Review by MarcDonald — APPROVED** [review]

**Comment by IvanGodinez21** [comment]
I found an error when I try to deploy a new Heroku app

Captura de pantalla 2021-05-30 171854

If you delete package-lock.json from the repository in master it should work @MarcDonald, I tested in other branch, this happens because Heroku use xubuntu, and I installed those packages from Windows 10 **cross your fingers**

**Comment by MarcDonald** [comment]
Yep, removed `package-lock.json` and it resolves the error. 

What would be the best way to create a package-lock that would work with Heroku? Generating one on an Ubuntu machine?

**Comment by IvanGodinez21** [comment]
I don't know exactly how Heroku installs the dependencies. But I think that it is the best solution for the moment

**Comment by MarcDonald** [comment]
Alright, I'll switch over to an Ubuntu machine and try it out

**Comment by MarcDonald** [comment]
Created a package-lock on Ubuntu and it seems to work now

**Comment by IvanGodinez21** [comment]
Yeah, sure

**Comment by IvanGodinez21** [comment]
Everything it's ok

Captura de pantalla 2021-05-30 175456

**Comment by MarcDonald** [comment]
Excellent
